src/source/to-shift.js

```diff
@@ -1,31 +1,98 @@
 'use strict';
 
-function removeFunctionContents(obj) {
-    obj.type = obj.type.replace(/^(?:Eager|Lazy)/, '');
+function tryRemoveFunctionContents(obj) {
```

Comment:
Could you document the `try` part? When does it fail? What does it do if it's not a function?

Reply:
Also, could you add an explanation or example of what this function does?

Reply:
"try" in the sense that it accepts non-functions too (and bails out if it's not a function node).